geps/gep-1709.md

```diff
@@ -253,6 +253,44 @@ move on to [certification](#certification) to report the results.
 [go]:https://go.dev
 [lib]:https://pkg.go.dev/sigs.k8s.io/gateway-api@v0.6.2/conformance/utils/suite
 
+### Gateway API version and channel
+
+The certification is related to a specific API version and a specific channel,
+therefore such information must be included in the final report.
+At test suite setup time, the conformance profile
+machinery gets all the CRDs with the field `.spec.group` equal to
+`gateway.networking.k8s.io`, and for each of them checks the annotations
+`gateway.networking.k8s.io/bundle-version` and
+`gateway.networking.k8s.io/channel`. If there are `CRD`s with different
+versions, the certification fails specifying that it's not possible to run the
+tests as there are different Gateway API versions installed in the cluster. If
+there are CRDs with different channels, the certification fails specifying
+that it's not possible to run the tests as there are different Gateway API
+channels installed in the cluster. If all the Gateway API `CRD`s have the same
+version and the same channel, the tests can be run and the detected version
+and channel will be set in the `GatewayAPIVersion` and `gatewayAPIChannel`
+fields of the final report. Furthermore, the suite must run all the
+experimental tests when the channel is `experimental`, and the related features
+are enabled.
+
+In addition to the `CRD`s version, the suite needs to check its version in
+relation to the `CRD`s one. To do so, a new `.go` file containing the current
+Gateway API version is introduced in the project and compiled with the
+conformance profile suite:
+
+```go
+const GatewayAPIVersion = "0.7.0"
+```
+
+At test suite setup time the conformance profile suite
+checks the `CRD`s version and the suite version; if the two versions differ,
+the certification fails.
+A new generator will be introduced in the project to generate the
```

Comment:
Tiny nit: The line lengths feel a bit weird here, I can't tell if 288 is intended to start a new paragraph, but it is not doing that when the markdown is rendered.